packages/plans-grid-next/src/hooks/data-store/use-per-month-description.tsx

```diff
@@ -0,0 +1,264 @@
+import {
+	isWpComFreePlan,
+	isWpcomEnterpriseGridPlan,
+	PLAN_BIENNIAL_PERIOD,
+	PLAN_ANNUAL_PERIOD,
+	PLAN_TRIENNIAL_PERIOD,
+	PlanSlug,
+	getPlanSlugForTermVariant,
+	TERM_ANNUALLY,
+	PLAN_HOSTING_TRIAL_MONTHLY,
+} from '@automattic/calypso-products';
+import { Plans } from '@automattic/data-stores';
+import { formatCurrency } from '@automattic/format-currency';
+import { useTranslate } from 'i18n-calypso';
+import type { GridPlan } from '../../types';
+
+interface UsePerMonthDescriptionProps {
+	siteId?: number | null;
+	planSlug: PlanSlug;
+	pricing: GridPlan[ 'pricing' ] | null;
+	isMonthlyPlan?: boolean;
+	storageAddOnsForPlan: GridPlan[ 'storageAddOnsForPlan' ];
+	coupon?: string;
+	useCheckPlanAvailabilityForPurchase: Plans.UseCheckPlanAvailabilityForPurchase;
+}
+
+export default function usePerMonthDescription( {
```

Comment:
Thanks for doing this refactor. Now that we are exposing this outside, I think best if we rename to `usePerMonthBillingDescription`. Although, it might actually be worth to generalise further into `useBillingDescription` (assuming that's accurate in the code).

Reply:
Sounds good!

Reply:
Renamed to usePlanBillingDescription by https://github.com/Automattic/wp-calypso/pull/91547/commits/f6d3e81bd9a3e9b51f1146b538b5863dc1da3970

Reply:
Even better! Thanks 🙇